docs/src/use/configure/migration-guide.md

```diff
@@ -0,0 +1,339 @@
+    }
+];
+```
+
+To support multiple configs for different glob patterns:
+
+```javascript
+// eslint.config.js
+export default [
+    "eslint:recommended", // Recommended config applied to all files
+    // File-pattern specific overrides
+    {
+        files: ["src/**/*", "test/**/*"],
+        rules: {
+                semi: ["warn", "always"]
+        }
+    },
+    {
+        files:["test/**/*"],
+        rules: {
+            "no-console": "off"
+        }
+    }
+    // ...other configurations
+];
+```
+
+### Configuring Language Options
+
+#### .eslintrc
+
+In `.eslintrc` files, you configure various language options across the `env`, `globals` and `parserOptions` properties.
+
+Global variables for specific runtimes (e.g `document` for browser JavaScript and `process` for Node.js ) are configured with the `env` property.
+
+```javascript
+// .eslintrc
+module.exports = {
+    env: {
+        browser: true,
+    },
+    globals: {
+        myCustomGlobal: "readonly",
+    },
+    parserOptions: {
+        ecmaVersion: 2022,
+        sourceType: "module"
+    }
+    // ...other config
+}
+```
+
+#### eslint.config.js
+
+In the `eslint.config.js`, the `globals`, and `parserOptions` are consolidated under the `languageOptions` key.
+
+There is no longer the `env` property in `eslint.config.js`. Global variables for specific runtimes are imported from the [globals](https://www.npmjs.com/package/globals) npm package and included in the `globals` property. You can use the spread operator (`...`)to import all globals:
+
+```javascript
+// eslint.config.js
+import globals from "globals";
+
+export default [
+    {
+        languageOptions: {
+            globals: {
+                ...globals.browser,
+                myCustomGlobal: "readonly"
+            },
+            parserOptions: {
+                ecmaVersion: 2022,
+                sourceType: "module"
+            }
+        }
+        // ...other config
+    }
+];
+```
+
+### Predefined Configs
+
+#### .eslintrc
+
+To use predefined configs, use the `extends` property:
+
+```javascript
+// .eslintrc.js
+module.exports = {
+    // ...other config
+    extends: "eslint:recommended",
+    rules: {
+        semi: ["warn", "always"]
+    },
+    // ...other config
+}
+```
+
+ESLint comes with two predefined configs:
+
+* -`eslint:recommended`: the rules recommended by ESLint
+* -`eslint:all`: all rules shipped with ESLint
+
+You can also use the `extends` property to extend a custom config. Custom configs can either be paths to local config files or npm package names:
+
+```javascript
+// .eslintrc.js
+module.exports = {
+    // ...other config
+    extends: ["eslint:recommended", "./custom-config.js", "eslint-config-my-config"],
+    rules: {
+        semi: ["warn", "always"]
+    },
+    // ...other config
+}
+```
+
+#### eslint.config.js
+
+`eslint.config.js` supports the built-in `eslint:recommended` and `eslint:all` configs, locally defined configs and npm package configs.
```

Comment:
the predefined configs are exported in `@eslint/js` now. https://github.com/eslint/eslint/tree/main/packages/js